pkg/httputil/httputil.go

```diff
@@ -0,0 +1,47 @@
+// Copyright 2017 clair authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// Package httputil implements some http fucntinality that is common for clair
+package httputil
+
+import (
+	"fmt"
+	"net/http"
+
+	"github.com/coreos/clair/pkg/version"
+)
+
+// Get creates a new GET httprequest and sets the User-Agnet to 'Clair/<version> (https://github.com/coreos/clair)'
+// If the response is not http.StatusOK, Get() will return an error with the response
+func Get(url string) (*http.Response, error) {
+	client := &http.Client{}
+
+	req, err := http.NewRequest("GET", url, nil)
+	if err != nil {
+		return nil, err
+	}
+
+	req.Header.Set("User-Agent", "Clair/"+version.Version+" (https://github.com/coreos/clair)")
+
+	resp, err := client.Do(req)
+	if err != nil {
+		return nil, err
+	}
+
+	if resp.StatusCode != http.StatusOK {
+		return nil, fmt.Errorf("Response was %d", resp.StatusCode)
```

Comment:
After playing with what this code would look like at the call site I think this is the best thing we can refactor this to look like:

```go
// Status2xx determines whether a request was successful or not.
func Status2xx(*http.Response) bool {
  return resp.StatusCode / 100 == 2
}

func GetWithUserAgent(url string) (*http.Response, error) {
  // No code for dealing with the Status here.
}
```

Now callers can do the following:

```go
resp, err := httputil.GetWithUserAgent(url)
if err != nil {
  return err
}
defer resp.Body.Close()

if !httputil.Status2xx(resp) {
  ...
}

...
```